tools/submit.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+if [[ $# < 1 ]]
+then
+    echo "Usage: ./submit.sh JOB [-t, --test], where JOB is the job script."
+    echo "Use option -t or --test to test without submitting job."
+    exit 1
+fi
+
+ERR=0
+
+BIN=$(sed -n 's/BIN\s*=\s*//p' < $1)
+CFG=$(sed -n 's/CFG\s*=\s*//p' < $1)
+
+if ! [[ -f $BIN ]]
+then
+   echo "Binary $BIN not found!"
+   exit 1
+fi
+
+if ! [[ -f $CFG ]]
+then
+   echo "Config $CFG not found!"
+   exit 1
+fi
+
+./check_vlasiator_cfg.sh $BIN $CFG > check.txt
+
+if grep -q "Invalid options" check.txt 
+then
+    echo "Invalid options in config file, check check.txt"
+    echo ""
+    ERR=1
+fi
+
+SW=$(sed -n s/^file_..\\s*=\\s*//p < $CFG)
+
+for i in $SW
+do
+    if ! test -f $i
+    then
+        echo "Solar wind file $i not found!"
+        echo ""
+        ERR=1
+    fi
+done
+
+MF=$(sed -n s/^atmosphericModelFile\\s*=\\s*//p < $CFG)
+if [[ -z $MF ]]
+then
+    MF=$(sed -n "s/ionosphere.atmosphericModelFile (=\(.*\))/\1/p" < check.txt)
+fi
+
+if ! test -f $MF
+then
+    echo "Atmospheric model file $MF not found!"
+    echo ""
+    ERR=1
+fi
```

Comment:
OK I see there's more of these environment-variable-based guys. So I suggest either of the below
- pass all input on the command line
- have the error messages print both the environment variable name and its contents – as I read this atm it'll print a blank if a variable is not set, it would be more helpful if it would tell the user what to set to define it. Me dumb user won't remember all these variables and I'll go sequentially setting them until it stops complaining.

If you pass them all on the command line as options, then you can check the count at the beginning and print what the options should be, in order.

If you want them as environment variables, how to skip one if it's not needed, e.g. in a 2D run no atmospheric model file is needed?

Reply:
doops